### ISSUE TITLE
assignment1/svm_classification

### ISSUE THREAD

**Issue by eli95** [Observerspy/CS231n#1]
# in class LinearClassifier: you define the default shape of X(D, N), and the shape of y_pred will be (N)
it is right.
def predict(self, X):
    """
    Use the trained weights of this linear classifier to predict labels for
    data points.

    Inputs:
    - X: D x N array of training data. Each column is a D-dimensional point.

    Returns:
    - y_pred: Predicted labels for the data in X. y_pred is a 1-dimensional
      array of length N, and each element is an integer giving the predicted
      class.
    """
    y_pred = np.zeros(X.shape[1])

but  in your assignment1/svm.ipynb:

y_train_pred = svm.predict(X_train)
print ('training accuracy: %f' % (np.mean(y_train == y_train_pred), ))
y_val_pred = svm.predict(X_val)
print ('validation accuracy: %f' % (np.mean(y_val == y_val_pred), ))

this time the shape of X_train is (N, D),  therefor y_pred will be (D, )?

FURTHERMORE:
the next step you do :
y_pred = np.argmax(X.dot(self.W), axis=1) 
totally  changed the shape of y_pred to (N,)

SO ,  WHY you add the line:
y_pred = np.zeros(X.shape[1])

thanks a lot ! you code and annotation really  illuminated me !

**Comment by Observerspy** [comment]
@ELI95 Hi, this sentence is above the **"TODO" block**, so  **it's provided by the class**. I don't think it has some meaning. Just as you said,  y_pred should be (N,).
